lib/iris/mesh/components.py

```diff
@@ -2724,73 +2741,7 @@ def __init__(
-        # Use node metadata, unless location is face/edge.
-        use_metadict = node_metadict.copy()
-        if location != "node":
-            # Location is either "edge" or "face" - get the relevant coord.
-            location_coord = self.mesh.coord(location=location, axis=axis)
-
-            # Take the MeshCoord metadata from the 'location' coord.
-            use_metadict = location_coord.metadata._asdict()
-            unit_unknown = Unit(None)
-
-            # N.B. at present, coords in a MeshXY are stored+accessed by 'axis', which
-            # means they must have a standard_name.  So ...
-            # (a) the 'location' (face/edge) coord *always* has a usable phenomenon
-            #     identity.
-            # (b) we still want to check that location+node coords have the same
-            #     phenomenon (i.e. physical meaning identity + units), **but** ...
-            # (c) we will accept/ignore some differences : not just "var_name", but
-            #     also "long_name" *and* "attributes".  So it is *only* "standard_name"
-            #     and "units" that cause an error if they differ.
-            for key in ("standard_name", "units"):
-                bounds_value = use_metadict[key]
-                nodes_value = node_metadict[key]
-                if key == "units" and (
-                    bounds_value == unit_unknown or nodes_value == unit_unknown
-                ):
-                    # Allow "any" unit to match no-units (for now)
-                    continue
-                if bounds_value != nodes_value:
-
-                    def fix_repr(val):
-                        # Tidy values appearance by converting Unit to string, and
-                        # wrapping strings in '', but leaving other types as a
-                        # plain str() representation.
-                        if isinstance(val, Unit):
-                            val = str(val)
-                        if isinstance(val, str):
-                            val = repr(val)
-                        return val
-
-                    nodes_value, bounds_value = [
-                        fix_repr(val) for val in (nodes_value, bounds_value)
-                    ]
-                    msg = (
-                        f"Node coordinate {node_coord!r} disagrees with the "
-                        f"{location} coordinate {location_coord!r}, "
-                        f'in having a "{key}" value of {nodes_value} '
-                        f"instead of {bounds_value}."
-                    )
-                    raise ValueError(msg)
-
-        # Don't use 'coord_system' as a constructor arg, since for
-        # MeshCoords it is deduced from the mesh.
-        # (Otherwise a non-None coord_system breaks the 'copy' operation)
-        use_metadict.pop("coord_system")
-
-        # Call parent constructor to handle the common constructor args.
+        points, bounds = self._load_points_and_bounds()
```

Comment:
I think you forgot to unpack the third value that is returned by `_load_points_and_bounds()`.